File: spring-data-requery-java/src/main/java/com/coupang/springframework/data/requery/repository/query/RequeryParametersParameterAccessor.java
package com.coupang.springframework.data.requery.repository.query;

import lombok.extern.slf4j.Slf4j;
import org.jetbrains.annotations.NotNull;
import org.springframework.data.repository.query.ParametersParameterAccessor;

/**
 * This class provides access to parameters of a user-defined method. It wraps ParametersParameterAccessor which catches
 * special parameters Sort and Pageable, and catches Arango-specific parameters e.g. AqlQueryOptions.
 *
 * @author debop@coupang.com
 * @since 18. 6. 8
 */
@Slf4j
public class RequeryParametersParameterAccessor extends ParametersParameterAccessor implements RequeryParameterAccessor {

    private final RequeryParameters parameters;


    public RequeryParametersParameterAccessor(@NotNull RequeryQueryMethod method, Object[] values) {
        super(method.getParameters(), values);
        this.parameters = method.getParameters();
    }

    @Override
    public RequeryParameters getParameters() {
        return parameters;
    }
}

File: spring-data-requery-kotlin/src/main/kotlin/org/springframework/data/requery/kotlin/repository/query/RequeryParametersParameterAccessor.kt
package org.springframework.data.requery.repository.query

import org.springframework.data.repository.query.ParametersParameterAccessor

/**
 * This class provides access to parameters of a user-defined queryMethod. It wraps ParametersParameterAccessor which catches
 * special parameters Sort and Pageable.
 *
 * @author debop@coupang.com
 */
class RequeryParametersParameterAccessor(override val parameters: RequeryParameters,
                                         values: Array<Any?>)
    : ParametersParameterAccessor(parameters, values), RequeryParameterAccessor {

    constructor(method: RequeryQueryMethod, values: Array<Any?>): this(method.parameters, values)

}